### PR TITLE
[SEMI-MODULAR]CWC Port from Beestation

### PR DESCRIPTION
**Pull request by Vizoplez**
## About The Pull Request
EXTREMELY WIP DRAFT

Ported from Beestation, credits to original authors and committers over there, love you guys.

- [x] Get a majority of code and sprites in
- [x] Get it functional
- [x] Bugtest and fix 

## Why It's Good For The Game
LETS FUCKING GOOOOOOOOOOOOO

## Changelog
:cl:
code: MASSIVE CLOCKWORK CULT REFACTOR, ALONG WITH UNDERLYING SUBSYSTEMS. WARNING.
add: Clockwork Cult. Credits to Beestation and their various contributors!
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
Credit for original author?

**Comment by Vizoplez** [comment]
> Credit for original author?

still setting up the PR and changelog itself. That comes after actually making this shit work.

**Comment by Vizoplez** [comment]
https://www.youtube.com/watch?v=dxiMX7nxr3Y me trying to kill the final integration error

**Comment by Vizoplez** [comment]
we do a little trolling, it seems.

**Comment by Vizoplez** [comment]
ready for testmerge

File: code/__DEFINES/~skyrat_defines/clockworkcult.dm [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+#define LIGHT_COLOR_CLOCKWORK 	"#BE8700"
+#define TELEPORT_MODE_CLOCKWORK 2
+#define COMSIG_ATOM_EMINENCE_ACT "atom_eminence_act"
+#define COMSIG_CLOCKWORK_SIGNAL_RECEIVED "clock_received"			//! When anything the trap is attatched to is triggered
+#define VANGUARD_TRAIT "vanguard"
+#define INCORPOREAL_MOVE_EMINENCE 4 //! same as jaunt, but lets eminence pass clockwalls
+#define STATUS_EFFECT_INTERDICTION /datum/status_effect/interdiction //! The affected is inside the range of an interdiction lens
+#define MOVESPEED_ID_INTERDICTION						"interdiction"
+
+#define ismovableatom(A) ismovable(A)
```
Comment (Gandalf2k15) [inline code/__DEFINES/~skyrat_defines/clockworkcult.dm:23]: Seems a but redundant?

Comment (Vizoplez) [reply code/__DEFINES/~skyrat_defines/clockworkcult.dm:23]: necessary for how its handled, the status effect itself doesnt actually have a speed defined, nor does it have a movespeed ID, which appears to be necessary; looking at other examples, and beestations example.

This is also used to display a HUD element, so the double define is necessary as they do separate things.

Comment (Gandalf2k15) [reply code/__DEFINES/~skyrat_defines/clockworkcult.dm:23]: Yeah but can't you just do ismovable instead of isatommovable?

**Comment by Gandalf2k15** [comment]
Yup, test merges fine, let's run a few tests of it.

**Comment by GoldenAlpharex** [comment]
You can't seem to scroll in the cog slab menu, from what I can see. Also, the transmissions between the servants seems to be in the wrong text format, it can't display apostrophes and the like.

There seems to be a good couple of sprites missing for the clothes, from what I've seen.
I don't know if it's intended, but there doesn't seem to be any kind of visual feedback when a cog has been installed in an APC? Unless it's meant to not be possible to be seen by ghosts.

**Comment by ghost** [comment]
You could TP into AI sat and get an easy victory

**Comment by Vizoplez** [comment]
> You could TP into AI sat and get an easy victory

Shouldnt be possible, code to prevent this. Will try and investigate why.

**Comment by Vizoplez** [comment]
> You can't seem to scroll in the cog slab menu, from what I can see. Also, the transmissions between the servants seems to be in the wrong text format, it can't display apostrophes and the like.
> 
> There seems to be a good couple of sprites missing for the clothes, from what I've seen.
> I don't know if it's intended, but there doesn't seem to be any kind of visual feedback when a cog has been installed in an APC? Unless it's meant to not be possible to be seen by ghosts.

You can and cant, its an issue with TGUI itself - and now there will be ways to see if an APC has a cog shoved into it.

**Comment by ghost** [comment]
> > You could TP into AI sat and get an easy victory
> 
> Shouldnt be possible, code to prevent this. Will try and investigate why.

I just tried it last round. in delta station....

**Comment by Vizoplez** [comment]
> > > You could TP into AI sat and get an easy victory
> > 
> > 
> > Shouldnt be possible, code to prevent this. Will try and investigate why.
> 
> I just tried it last round. in delta station....

I said should not, not is not.

**Comment by Vizoplez** [comment]
linters randomly crashing, maintainers please re-run

File: code/__DEFINES/~skyrat_defines/clockworkcult.dm (hunk 1 above)
Comment (Vizoplez) [reply code/__DEFINES/~skyrat_defines/clockworkcult.dm:23]: i dont know probably

**Comment by Vizoplez** [comment]
the unit test is going schizo mode, please ignore for now.

Edit: Yeah, it wont echo the failure locally. Bruh.

**Comment by Vizoplez** [comment]
Fixed and ready to testmerge. Reebe now loads properly.

**Comment by Gandalf2k15** [comment]
Okay instead of making this permanent spawn, make it use the away mission map loader if it's needed

**Comment by Vizoplez** [comment]
> Okay instead of making this permanent spawn, make it use the away mission map loader if it's needed

I dont think this supports station traits but i'll try.

**Comment by Vizoplez** [comment]
I lack the energy or motivation to find another workaround for loading issues, along with fixing the few last issues. Feel free to Re-PR this and finish it, its almost completely functional; and currently does work - if its slightly heavier on memory.

**Comment by Gandalf2k15** [comment]
I'll do this, leave it open.

**Comment by Gandalf2k15** [comment]
I AM CULTIVATING ALL THE MEMORY

**Comment by Gandalf2k15** [comment]
I'm just gonna send it.

**Comment by GoldenAlpharex** [comment]
Uh, you unticked a good couple of clockwork-related files, and removed some Reebe stuff, was that intentional?
Also, maybe you could try to work something out so that there's less space z-levels when the game would load up Reebe, so it doesn't affect the main server since Clockwork Cult won't really be a thing there?

**Comment by Gandalf2k15** [comment]
Yep

**Comment by Gandalf2k15** [comment]
> Uh, you unticked a good couple of clockwork-related files, and removed some Reebe stuff, was that intentional?
> Also, maybe you could try to work something out so that there's less space z-levels when the game would load up Reebe, so it doesn't affect the main server since Clockwork Cult won't really be a thing there?

We dont really need 7 space ruin z levels

**Comment by GoldenAlpharex** [comment]
> > Uh, you unticked a good couple of clockwork-related files, and removed some Reebe stuff, was that intentional?
> > Also, maybe you could try to work something out so that there's less space z-levels when the game would load up Reebe, so it doesn't affect the main server since Clockwork Cult won't really be a thing there?
> 
> We dont really need 7 space ruin z levels

We don't /need/ them, but it just cuts off a lot of content for space exploration for something that wouldn't even be seen on the main server, no? Could be a config option I guess.

**Comment by unit0016** [comment]
> > > Uh, you unticked a good couple of clockwork-related files, and removed some Reebe stuff, was that intentional?
> > > Also, maybe you could try to work something out so that there's less space z-levels when the game would load up Reebe, so it doesn't affect the main server since Clockwork Cult won't really be a thing there?
> > 
> > 
> > We dont really need 7 space ruin z levels
> 
> We don't /need/ them, but it just cuts off a lot of content for space exploration for something that wouldn't even be seen on the main server, no? Could be a config option I guess.

It doesn't cut the content off, they're still spending the same space ruin budget. In fact, a lot of space'll just now be.. very uncomfortably close to each other.

**Comment by Vizoplez** [comment]
https://youtu.be/Bu_hw963AG4?t=12

**Comment by Gandalf2k15** [comment]
Ues

**Comment by Gandalf2k15** [comment]
Is this xompatablw with dynamic

**Comment by Vizoplez** [comment]
> Is this xompatablw with dynamic

not yet

**Comment by Vizoplez** [comment]
what the fuck grille error

**Comment by Vizoplez** [comment]
ok im based dynamic ready I guess I havent actually tested this

**Comment by Gandalf2k15** [comment]
testing again

**Comment by Nerev4r** [comment]
#5524 broke this.

**Comment by Vizoplez** [comment]
> #5524 broke this.

yes i know

**Comment by Vizoplez** [comment]
wOoWOOO

Closed until I can find the effort to follow new dev guidelines involving this

**Pull request closed without merging**